app/source/acta.cpp

```diff
@@ -44,12 +38,12 @@ Result ACTA_CreateLocalAccount() {
 	return (Result)cmdbuf[1];
 }
 
-Result ACTA_ResetAccount(u8 account_index, bool format_nnid) {
+Result ACTA_ResetAccount(Account account_index, bool format_nnid) {
```

Comment:
Account index is gotten by frd functions though, which uses pretty much the same account struct (just without the undefined due to me using enum classes over enums)

Reply:
i will remove the definition for dev though as its redundant

Reply:
it creates an frd account with friend account id, uses friend account id to switch act/frd accounts, then resets the act account, using friend account id

and account is nascenvironment + 1, which doesnt get applied anywhere really, except to set the environment (account - 1), which is used for createlocalaccount, which originally just used 2 and 1 (Pretendo and NASC_Env_Test), which is the only valid thing it could really be set to

so ehhhhhhh
idk

Reply:
the account ID of a new account can be defined by the caller (on ACT this is limited to 8 accounts, not sure on FRD), so you can apply any environment to any account ID you create. For example you could have the following accounts in the console:

- Account ID 1 = Prod
- Account ID 2 = Prod
- Account ID 3 = Dev
- Account ID 4 = Test

Or you can use whatever combination you want. Setting the `account ID` to `NASCEnvironment + 1` on nimbus is done for convenience

Anyways, I'm looking at this by the perspective on how the account system (probably) works internally. We can choose also to not handle any more than two accounts on nimbus and leave this how it is right now.

Reply:
ok, i set it back to u8

Reply:
I meant this for the ACT and FRD functions (or at least the ACT ones)